docs/src/main/paradox/code-details.md

```diff
@@ -0,0 +1,78 @@
+# Understanding the code
+
+Adding gRPC support to a vanilla Play application requires a few steps:
+
+### 1. `sbt-akka-grpc` 
+
+Add the Akka gRPC plugin on `project/plugins.sbt` 
+
+```scala
+// Akka GRPC
+addSbtPlugin("com.lightbend.akka.grpc" %% "sbt-akka-grpc" % "0.4.2")
+```
+
+and enable it on your project (in `build.sbt`):
+
+```scala
+lazy val `play-scala-grpc-example` = (project in file("."))
+  .enablePlugins(PlayScala)
+  .enablePlugins(AkkaGrpcPlugin) // enables source generation for gRPC
+  .enablePlugins(PlayAkkaHttp2Support) // enables serving HTTP/2 and gRPC
+``` 
+
+The `AkkaGrpcPlugin` locates the gRPC `.proto` files and generates source code from it. Remember to enable the plugin 
+in all the projects of your build that want to use it.
+
+Note how the `PlayAkkaHttp2Support` is also enabled. gRPC requires HTTP/2 transport and Play supports it only as an opt-in plugin.
+
+
+### 2.a Serving (Akka) gRPC Services
+
+Have a look at the `conf/routes` file where you'll notice how to embed a gRPC router within a normal play application. 
+You can in fact mix normal Play routes with gRPC routers like this to offer a mixed service. You'll notice that we 
+bind the `/` path to the `controllers.HomeController` like usual route,
+and then we use the `->` router binding syntax to bind the `routers.HelloWorldRouter`. This is because gRPC services 
+have paths correspond to their "methods", yet this is handled by its internal infrastructure and end-users need
+not concern themselves about the exact names – clients too are generated from the appropriate 
+`app/protobuf/helloworld.proto` file after all.
+
+You can read more about [Service gRPC from a Play App](https://developer.lightbend.com/docs/play-grpc/current/serving-grpc.html) in the docs.
+
+### 2.b Injecting Akka-gRPC Clients 
+
+Similarily to the server side, the sources are generated by the Akka gRPC plugin by having it configured to emit the client as well:
+
+```
+// build.sbt
+akkaGrpcExtraGenerators += PlayScalaClientCodeGenerator,
+``` 
+
+In order to make the gRPC clients easily injectable, we need to enable the following module in Play as well (in this 
+example app this has been done already though):
+
+```
+// application.conf
+play.modules {
+  # To enable Akka gRPC clients to be @Injected
+  enabled += example.myapp.helloworld.routers.AkkaGrpcClientModule
+}
+```
+
+Which in turn allows us to inject clients to any of the services defined in our `app/proto` directory, just like so:
+
+```scala
+import akka.stream.Materializer
+import example.myapp.helloworld.grpc.{HelloReply, HelloRequest}
+import javax.inject.Inject
+
+import scala.concurrent.Future
+
+
+class HomeController @Inject() (mat: Materializer, greeterServiceClient: GreeterServiceClient) extends InjectedController {
+  implicit val ec = mat.executionContext
```

Comment:
Why not inject the `ExecutionContext` instead?

Reply:
Ah, forgot to fix the docs. This was copy/pasted from the code and I had it fixed there.
Actually, I'll make this snippet injected via paradox code embedding.